src/storage/btree.c

```diff
@@ -19381,12 +19661,38 @@ btree_range_opt_check_add_index_key (THREAD_ENTRY * thread_p, BTREE_SCAN * bts,
       db_make_null (&new_key_value[i]);
     }
 
-  for (i = 0; i < multi_range_opt->num_attrs; i++)
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+  if (bts->is_cur_key_compressed)
     {
-      error = pr_midxkey_get_element_nocopy (new_mkey, multi_range_opt->sort_att_idx[i], &new_key_value[i], NULL, NULL);
-      if (error != NO_ERROR)
+      DB_MIDXKEY *prefix_mkey = NULL;
+      int idx;
+      assert (bts->common_prefix_size > 0);
+
+      check_validate (bts);
+      prefix_mkey = db_get_midxkey (&(bts->common_prefix_key));
```

Comment:
How about using `&(bts->common_prefix_key)` directly?

Reply:
It was explained that this was to avoid repeated calls within loop statements.
I also found out that to change it, I had to put `db_get_midxkey (&(bts->common_prefix_key))`, not `&(bts->common_prefix_key)`.